pkg/commands/docker.go

```diff
@@ -93,12 +95,34 @@ func (t *dockerBuildTask) Build(runner build.Runner) error {
 		args = append(args, "--build-arg", buildArg)
 	}
 
+	for _, buildSecretArg := range t.buildSecretArgs {
+		args = append(args, "--build-arg", buildSecretArg)
+	}
+
 	if t.contextDir != "" {
 		args = append(args, t.contextDir)
 	} else {
 		args = append(args, ".")
 	}
-	return runner.ExecuteCmd("docker", args)
+
+	return runner.ExecuteCmdWithObfuscation(func(args []string) {
+		if len(t.buildSecretArgs) > 0 {
```

Comment:
Conceptually I'd like to have a substitution map so we can reuse any obfuscation work and lookups in multi step builds later.